allennlp/run.py

```diff
@@ -23,6 +23,10 @@ def run():
 if __name__ == "__main__":
     # First, let Pytorch's multiprocessing module know how to create child processes.
     # Refer https://docs.python.org/3.7/library/multiprocessing.html#multiprocessing.set_start_method
-    torch.multiprocessing.set_start_method("spawn")
+    # We use force=True because this is guarded by the if __name__ ... block, which
+    # means that it only runs if allennlp is being run as a binary.
+    # Without this, if another library that allennlp uses has set the start method,
+    # this line would crash.
+    torch.multiprocessing.set_start_method("spawn", force=True)
```

Comment:
Hmm, this is interesting. What are the implications if another library has set a different start method and we force past that? I'm having trouble finding the docs for `force`. Pytorch seems to import it from Python which is here: https://github.com/python/cpython/blob/master/Lib/multiprocessing/context.py#L241 But the docs don't mention it and the code doesn't help much either... (Docs : https://docs.python.org/3/library/multiprocessing.html#multiprocessing.set_start_method doesn't mention it.)

Reply:
I'm guessing you have some other source that explains why it's necessary?

Reply:
It appears that setting this is some kind of global python state, which I don't understand - e.g running this sequentially in two separate python interpreters one after another, closing inbetween, results in it crashing. I'm not super sure about the implications, but using distributed with pytorch does require us to use this spawn method.

Reply:
Actually on my linux machine, even opening a python interpreter causes it to already be set.

Reply:
Oof, this is nasty. What it sounds like is that some orphaned processes are keeping alive the process that Python creates to spawn the workers (or something in that vein). See https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods.